hashicorp/dev-portal
# feat: add tabs nested styling

## 🔗 Relevant links

- [Preview link][/waypoint/tutorials/get-started-docker/get-started-install] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Implements style for nested tabs in `tutorials` and `docs` MDX content.

## 🤷 Why

To implement tab styles to match our design specs.

## 🛠️ How

- Adds a prop `allowNestedStyles` to `components/tabs`
- Implements nested tab styles when `allowNestedStyles` is set to `true`
- Sets up shared `MdxTabs` and `MdxTab` in `components/dev-dot-content`, to ensure identical behaviour and appearance in docs and learn
- In `MdxTabs` wrapper, enables `allowNestedStyles`

## 📸 Design Screenshots

<details>

<summary>Full design spec</summary>

<img width="1000" alt="Nested tabs" src="https://user-images.githubusercontent.com/4624598/173360604-9e987364-a427-4a4b-94e0-8eec0e62b1ae.png">


</details>

## 🧪 Testing

- [ ] Visit a tutorial page with nested tabs, such as [/waypoint/tutorials/get-started-docker/get-started-install][/waypoint/tutorials/get-started-docker/get-started-install]
    - Ensure the tabs appear as expected
- [ ] Visit a tutorial page with normal tabs
    - Ensure normal, non-nested tabs appear as expected
- [ ] Visit a docs page with normal tabs, such as [/vault/api-docs/secret/rabbitmq][/vault/api-docs/secret/rabbitmq]
    - Ensure normal, non-nested tabs appear as expected
- [ ] Visit a page with non-MDX use of tabs, such as [/waypoint/install][/waypoint/install]
    - Ensure the appearance and functionality of tabs is the same as before

## Anything else

- Note on removal of `defaultActiveIdx` in docs `Tabs`
    - The `defaultActiveIdx` property will not be supported for `Tabs` in dev-dot
    - This feels like a safe change to make, it seems there are [zero uses of this property in docs content according to SourceGraph](https://hashicorp-mktg.sourcegraph.com/search?q=context:global+file:.mdx%24+defaultTabIdx&patternType=literal).

[task]: https://app.asana.com/0/1202097197789424/1202151178802272/f
[preview]: https://dev-portal-git-zsstyle-nested-tabs-hashicorp.vercel.app
[/waypoint/tutorials/get-started-docker/get-started-install]: https://dev-portal-git-zsstyle-nested-tabs-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-install
[/vault/api-docs/secret/rabbitmq]: https://dev-portal-git-zsstyle-nested-tabs-hashicorp.vercel.app/vault/api-docs/secret/rabbitmq
[/waypoint/install]: https://dev-portal-git-zsstyle-nested-tabs-hashicorp.vercel.app/waypoint/install
